### DISCUSSION TITLE
Easing parameter selection for FFM

### DISCUSSION THREAD

**Discussion by jpfeuffer** [OpenMS/OpenMS#5388] [Ideas]
Optimizing parameters for FFM is a huge pain. What can we do to ease this?
Especially the noise and FWHM parameters have such a huge impact on both quality and runtime such that it is almost impossible to figure out good parameters without looking at the data for a long time.
Especially @oliveralka @FabianAicheler @timosachsenberg

**Comment by jpfeuffer** [comment]
It is not even clear what chrom_fwhm means. What is "expected"? The average as in expected value? How does it influence the trace finding exactly?
Especially since there is a min and max_trace_length param also.

**Comment by tjeerdijk** [comment]
Hi Julianus,

For a complicated optimisation problem like this (parameter optimisation for FFM) we could try a Gaussian Process - Active Learning approach as detailed in this book: https://bobby.gramacy.com/surrogates/.

Maybe we can write out a Master Thesis Project for the machine learners?

Take care from Tjeerd

**Comment by timosachsenberg** [comment]
Multiplex or Metabo?

**Reply by jpfeuffer** [reply]
Metabo. Which is also very very slow. I am currently working on an indexing structure for mz ranges.

**Reply by jpfeuffer** [reply]
I profiled and findNearest takes over 50% of the runtime.

**Reply by timosachsenberg** [reply]
We already have kD tree but I guess you know that one. In my experience FFMetabo was mainly slow if accidentally profile data was provided or if the trace is always stopped after collecting few peaks resulting in many noise related peaks. Are you using a release build?

**Reply by jpfeuffer** [reply]
I double-checked with @oliveralka.
The data looks fine and is centroided. There are just too many peaks on a 2h gradient for this type of lookup.

We actually only need indexing in mz range, so kdtree was overkill. For RT only neighboring scans are used.
But it did not make any difference, although the range to search decreased significantly. I think there are just millions of apexes that it wants to search. It takes hours.
